### PR TITLE
Rewrite l10n system

### LINKED ISSUE #81: [l10n] multiple messages in the same tag
# Output multiple messages at once

## Describe the feature

Currently, to output multiple messages at the same time, you need two tags, like this:
```handlebars
{{ #message }}key1{{ /message }} {{ #message }}key2{{ /message }} {{ #message }}key3{{ /message }}
```
and so on.
That can get very long, very fast.

I propose being able to do it like this:
```handlebars
{{ #message }}key1;key2;key3{{ /message }}
```
and so on.

## Reasons to add this feature

Less typing, less repetition

## Alternatives to this feature

What we do now

## Additional information, if any

As you can with triple pipes and square brackets currently, you should be able to escape the semicolons with backslashes.

The old method should be identical to the new one in all aspects, meaning that placeholders should still work. In addition, each should be joined with a space automatically.

The splitter symbol can be something other than a semicolon if you prefer.

### PR DESCRIPTION
**Pull request by RedGuy12**
# Rewrite l10n system

## Resolves

below

### Tech Debt
* Fix prettier formatting for locale files
* Move all custom properties on `request` related to l10n to `request.localization`
* Expand abbreviations
* Rewrite many parts of l10n.js

### String updates
* Split longer strings into shorter ones
* Localise some branding

### Bugs
* Fix `/old` being mostly blank

### Features 
* Support multiple messages in the same tag (Resolves #81)
* Support nested properties
* Support nested messages being anywhere in the variable
* Export the "all languages" variable to prepare for #108

### Reason for changes

* @wgyt said l10n.js was too confusing
* I wanted to keep all l10n fields together
* Bugs are bugs and features are features
* Shorter strings are good because if you need to change part of it, only that part gets re-translated

## Tests

yes, mostly works but
* localhost_3000_ (2)
* the title on all pages (expect docs and home) is `title`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -44,26 +44,13 @@ app.use((request, response, next) => {
...
-app.use(errorPages);
-
-app.use(
-	express.static(path.resolve(directory, "static"), {
-		dotfiles: "allow",
-		immutable: true,
-		maxAge: 31536000,
-	}),
-);
-
```
Comment (wgyt) [inline src/index.js:56]: why do we not need this?

File: src/main/index.js [hunk 2]
```diff
@@ -82,10 +82,10 @@ app.use((_, response, next) => {
 			 */
 			(index) => {
 				const code = codeblocks.eq(index),
-					[languageClass, language = "plaintext"] =
-						/lang(?:uage)?-(?<language>\w+)/u.exec(code.attr("class") || "") ?? [];
+					[fullClass, language = "plaintext"] =
```
Comment (wgyt) [inline src/main/index.js:85]: nice var name change

**Review by wgyt — COMMENTED** [review]
this was a LOT to go through, here are my questions and comments. Thanks @RedGuy12

File: .github/workflows/install.yml [hunk 3]
```diff
@@ -37,18 +37,23 @@ jobs:
               run: npm audit fix
 
             - name: Unneeded
+              if: ${{ !cancelled() }}
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply .github/workflows/install.yml:40]: `npm audit fix` will crash if it couldn't fix all vulns. Normally that would stop the action, but I use `if: ${{ !cancelled() }}` to say "run it anyway"

[the remark this answers is not in the document]
Comment (wgyt) [reply .github/workflows/install.yml:40]: oh ok

File: .prettierignore [hunk 4]
```diff
@@ -9,7 +9,7 @@ plugin_packages/
 .nvm
 .eslintcache
 .node_repl_history
-_locales
-!locales/en_US.json
+_locales/*.json
+!_locales/en_US.json
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply .prettierignore:13]: if i format the code, tx overrides formating of language files other than the source strings.

[the remark this answers is not in the document]
Comment (wgyt) [reply .prettierignore:13]: oh ok

File: _locales/en_US.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
 	"emailSubject": { "string": "1Auth Email Verification" },
 	"emailCodeIs": { "string": "Your code is {0}." },
 	"emailNoShare": { "string": "Don’t share it with anyone else." },
-	"emailNoShareWhy":{"string": "Otherwise, they could be able to act on your behalf."},
+	"emailNoShareWhy": { "string": "Otherwise, they could be able to act on your behalf." },
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply _locales/en_US.json:10]: yes but if you know someone's code and email, you can authenticate as them pretty easily.

File: PAGE_TEMPLATE.html [hunk 6]
```diff
@@ -3,7 +3,7 @@
 	<head>
 		<meta charset="utf-8" />
 		<meta name="viewport" content="width=device-width, initial-scale=1" />
-		<title>{{ title }} | 1Auth</title>
+		<title>{{ #message }}title|||{{ title }}{{ /message }}</title>
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply PAGE_TEMPLATE.html:6]: `{{ #message }}` is a custom thing i made to render translations

File: package.json [hunk 7]
```diff
@@ -26,7 +26,7 @@
 		"lint:mustaches": "node --trace-warnings --harmony .github/workflows/mustaches.js",
 		"lint:typescript": "tsc --project jsconfig.json",
 		"start": "node --trace-warnings --harmony src/index.js",
-		"build:tailwind": "postcss src/tailwind.sass -o src/main/style.css"
+		"build:tailwind": "postcss src/tailwind.sass --output src/main/style.css"
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply package.json:29]: why not :p

i like to be descriptive

File: package.json [hunk 8]
```diff
@@ -123,6 +123,7 @@
 		"@ultraq/icu-message-formatter": "^0.11.1",
 		"accepts": "^1.3.7",
 		"atob": "^2.1.2",
+		"balanced-match": "^1.0.2",
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply package.json:126]: a new lib used in l10n.js

File: package.json [hunk 9]
```diff
@@ -196,15 +198,15 @@
...
 			"css-mqpacker": {},
 			"cssnano": {
 				"autoprefixer": false,
 				"discardUnused": true,
 				"mergeIdents": true,
 				"reduceIdents": true,
 				"zindex": true
-			}
+			},
+			"autoprefixer": {}
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply package.json:209]: autoprefixer adds vendor prefixes. but it wasn't properly prefixing the css so i moved it lower in the list

File: src/auth/email/index.js [hunk 10]
```diff
@@ -82,11 +83,14 @@ const client = {
 								{
 									code,
 
-									message: mustacheFunction(request.languages, request.messages),
+									message: mustacheFunction(
+										request.localization.languages,
+										request.localization.messages,
+									),
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply src/auth/email/index.js:89]: yes

File: src/index.js (hunk 1 above)
Comment (RedGuy12) [reply src/index.js:56]: it was duplicated

File: src/auth/github/index.js [hunk 11]
```diff
@@ -34,8 +34,7 @@ const client = {
 					body:
 						"client_id=7b64414fe57e07d1e969" +
 						`&client_secret=${process.env.githubSECRET}` +
-						`&code=${request.query?.code}` +
-						`&state=${request.query?.state}`,
+						`&code=${request.query?.code}`,
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply src/auth/github/index.js:37]: i don't remember why i did this

**Review by wgyt — APPROVED** [review]

**Comment by RedGuy12** [comment]
fixed bugs

**Pull request merged**